src/modules/ekf2/EKF2.cpp

```diff
@@ -2641,32 +2714,41 @@ int EKF2::task_spawn(int argc, char *argv[])
 
 			vehicle_status_sub.update();
 
-			for (uint8_t mag = 0; mag < mag_instances; mag++) {
-				uORB::SubscriptionData<vehicle_magnetometer_s> vehicle_mag_sub{ORB_ID(vehicle_magnetometer), mag};
+			uint8_t selected_mag = 0;
+			uint8_t selected_imu = 0;
 
-				for (uint8_t imu = 0; imu < imu_instances; imu++) {
+			for (uint8_t mag = 0; mag < valid_mag_instances; mag++) {
```

Comment:
same as before, if `valid_mag_instances` > `mag_instances` you'll create more instances than required.

Reply:
In this case, the limited number of instances can't be used instead of valid_*_instances as the sensor might not be advertising and multiple sensors might have to be checked in order to create a valid instance. Some checks have to be added

Reply:
A return clause has been added to prevent excessive spawns